src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py

```diff
@@ -12,28 +12,24 @@
 
 def collect_blobs(blob_service, container, pattern=None):
     """
-    List the blobs in the given blob container, filter the blob by comparing their path to the given
-    pattern.
+    List the blobs in the given blob container, filter the blob by comparing their path to the given pattern.
     """
-    # TODO: deal with container which contains more that 5000 files.
-    #       In that case, the listing operation will page the results.
-
     if not blob_service:
         raise ValueError('missing parameter blob_service')
 
     if not container:
         raise ValueError('missing parameter container')
 
     if not _pattern_has_wildcards(pattern):
-        return [pattern]
+        return [pattern] if blob_service.exists(container, pattern) else []
```

Comment:
This is the only meaningful change. By ensuring the pattern match one blob otherwise returns an empty list, this prevents any error of missing blobs from happening.